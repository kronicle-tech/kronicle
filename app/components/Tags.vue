<template>
  <ul v-if="tags && tags.length > 0">
    <li v-for="(tag, tagIndex) in tags"
        :key="tagIndex"
    >
      {{ tag.key }}<span v-if="tag.value !== undefined">: {{ tag.value }}</span>
    </li>
  </ul>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import { Tag } from "~/types/kronicle-service";

export default Vue.extend({
  props: {
    tags: {
      type: Array as PropType<Tag[]>,
      default: undefined,
    },
  },
})
</script>
