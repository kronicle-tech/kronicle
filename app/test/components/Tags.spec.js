import { mount } from '@vue/test-utils'
import Tags from '@/components/Tags.vue'

describe('Tags', () => {
  let propsData
  let wrapper
  const createWrapper = () => {
    wrapper = mount(Tags, { propsData })
  }

  beforeEach(() => {
    propsData = {}
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when tags prop is not set', () => {
    test('renders nothing', () => {
      createWrapper()
      expect(wrapper.html()).toEqual(``)
    })
  })

  describe('when tags prop is set to an empty array', () => {
    beforeEach(() => {
      propsData.tags = []
    })

    test('renders nothing', () => {
      createWrapper()
      expect(wrapper.html()).toEqual(``)
    })
  })

  describe('when tags prop is set to one tag', () => {
    beforeEach(() => {
      propsData.tags = [
        {
          key: 'test-tag-key-1',
        },
      ]
    })

    test('renders an unordered list showing the tag', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when tags prop is set to one tag with a value', () => {
    beforeEach(() => {
      propsData.tags = [
        {
          key: 'test-tag-key-1',
          value: 'test-tag-value-1',
        },
      ]
    })

    test('renders an unordered list showing the tag', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when tags prop is set to multiple tags', () => {
    beforeEach(() => {
      propsData.tags = [
        {
          key: 'test-tag-key-1',
        },
        {
          key: 'test-tag-key-2',
        },
        {
          key: 'test-tag-key-3',
        },
      ]
    })

    test('renders an unordered list showing the tags', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
