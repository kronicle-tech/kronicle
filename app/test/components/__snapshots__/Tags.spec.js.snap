// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags when tags prop is set to multiple tags renders an unordered list showing the tags 1`] = `
"<ul>
  <li>
    test-tag-1
  </li>
  <li>
    test-tag-2
  </li>
  <li>
    test-tag-3
  </li>
</ul>"
`;

exports[`Tags when tags prop is set to one tag renders an unordered list showing the tag 1`] = `
"<ul>
  <li>
    test-tag-1
  </li>
</ul>"
`;
